src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -1084,10 +1089,12 @@ class spell_dk_will_of_the_necropolis : public SpellScriptLoader
 
             bool CheckProc(ProcEventInfo& eventInfo)
             {
-                if (GetTarget()->HasAura(SPELL_DK_WILL_OF_THE_NECROPOLIS))
+                Unit* actionTarget = eventInfo.GetActionTarget();
+
+                if (actionTarget->HasAura(SPELL_DK_WILL_OF_THE_NECROPOLIS))
                     return false;
 
-               return GetTarget()->HealthBelowPctDamaged(30, eventInfo.GetDamageInfo()->GetDamage());
+                return actionTarget->HealthBelowPctDamaged(30, eventInfo.GetDamageInfo()->GetDamage());
```

Comment:
gettarget was good here, and pls replace the healthpct value with effect_0 value